### PR TITLE
executor: migrate `trace_test.go` to testify

### LINKED ISSUE #28622: migrate test-infra to testify for `trace_test.go`

### PR DESCRIPTION
**Pull request by johnhaxx7**
### What problem does this PR solve?

Issue Number: close #28622

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: executor/trace_test.go [hunk 1]
```diff
@@ -67,12 +71,14 @@ func rowsOrdered(rows [][]interface{}) bool {
...
+	tk := testkit.NewTestKit(t, store)
 	tk.MustExec("use test")
 	tk.MustExec("create table tp123(id int);")
 	rows := tk.MustQuery("trace plan select * from tp123").Rows()
-	c.Assert(rows, HasLen, 1)
-	c.Assert(rows[0], HasLen, 1)
-	c.Assert(rows[0][0].(string), Matches, ".*zip")
+	require.Len(t, rows, 1)
+	require.Len(t, rows[0], 1)
+	require.Regexp(t, ".*zip", rows[0][0].(string))
```
Comment (hawkingrei) [inline executor/trace_test.go:83]: ```.(string)``` can be removed.

Comment (johnhaxx7) [reply executor/trace_test.go]: nice catch! thanks!

**Review by hawkingrei — APPROVED** [review]
LGTM

**Review by tisonkun — APPROVED** [review]

**Pull request merged**